src/data/matrix/basic.lean

```diff
@@ -607,6 +607,16 @@ instance semiring.smul_comm_class [fintype n] [monoid R] [distrib_mul_action R 
   (λ i j, a * M i j) ⬝ N = a • (M ⬝ N) :=
 smul_mul a M N
 
+lemma smul_pow [fintype n] [decidable_eq n] [monoid R] [distrib_mul_action R α]
+  [is_scalar_tower R α α] [smul_comm_class R α α]
+  (k : R) (A : matrix n n α) (p : ℕ) :
+  (k • A) ^ p = k ^ p • A ^ p :=
```

Comment:
This is true more generally as
```suggestion
lemma smul_pow [fintype n] [decidable_eq n] [monoid R] [distrib_mul_action R α]
  [is_scalar_tower R α α] [smul_comm_class R α α]
  (k : R) (A : α) (p : ℕ) :
  (k • A) ^ p = k ^ p • A ^ p :=
```
for which exactly the same proof works.

This lemma should probably be near `smul_mul_smul`, although perhaps powers aren't yet developed there.